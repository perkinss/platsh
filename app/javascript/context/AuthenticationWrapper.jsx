import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
