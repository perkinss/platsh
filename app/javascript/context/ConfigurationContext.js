import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
