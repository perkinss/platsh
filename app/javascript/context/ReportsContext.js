import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
