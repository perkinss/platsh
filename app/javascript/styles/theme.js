import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
