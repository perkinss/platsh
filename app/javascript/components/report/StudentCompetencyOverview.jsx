import React from 'react'
import { Typography, Grid } from '@material-ui/core'
import { useTheme } from '@material-ui/styles'
import RadialChart from "./RadialChart";
import CircularProgress from "@material-ui/core/CircularProgress";

export default function StudentCompetencyOverview(props) {
    const {student, course, classes, competencyData, competencyLabels, loading, graphLabels } = props
    const theme = useTheme()

    let barSize, chartHeight, nameLabelOffset,labelFontSize, radialBarOffset,valueFontSize
    if (window.innerWidth < 700) {
        barSize = 150
        chartHeight = 280
        labelFontSize = '20px'
        nameLabelOffset = 120
        radialBarOffset = 0
        valueFontSize = '20px'
    } else {
        barSize = 200
        chartHeight = 400
        labelFontSize = '25px'
        nameLabelOffset = 185
        radialBarOffset = 0
        valueFontSize = '30px'
    }

    return (
        <div style={{paddingTop: theme.spacing(5)}}>
            <Typography variant={'h5'}>Competency Overview</Typography>
            {loading && <CircularProgress color={'primary'} />}
            {!loading && competencyData &&
            <RadialChart
                classes={classes}
                key={`chart-competency-overview-${student.name}`}
                labels={competencyLabels}
                series={competencyData}
                height={chartHeight}
                radialBarSize={barSize}
                radialBarOffset={-10}
                nameLabelOffset={nameLabelOffset}
                hollowSize={'22%'}
                valueFontSize={valueFontSize}
                labelFontSize={labelFontSize}
                showToolbar={true}
            />}
        </div>

    )

}