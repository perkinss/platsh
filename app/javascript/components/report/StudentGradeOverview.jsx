import React, { useEffect, useState } from 'react'
import { Typography, Grid } from '@material-ui/core'
import { useTheme } from '@material-ui/styles'
import CircularProgress from "@material-ui/core/CircularProgress";
import Paper from "@material-ui/core/Paper";

export default function StudentGradeOverview(props) {
    const { competencyTotal, contentTotal, competenciesLoading, contentLoading, courseContentWeighting } = props
    const theme = useTheme()

    const competencyLabel = competencyTotal ? competencyTotal.toFixed(0) +  ' %' : 'N/A'
    const contentLabel = contentTotal ? contentTotal.toFixed(0) +  ' %' : 'N/A'
    const [courseTotalLabel, setCourseTotalLabel] = useState('')

    const calcCourseTotal = (contentTotal, competencyTotal, topicsWeighting) => {
        if (!topicsWeighting) {
            return ((contentTotal + competencyTotal) / 2)
        } else {
            const competencyWeighting = 100 - topicsWeighting
            return ((contentTotal * topicsWeighting) + (competencyTotal * competencyWeighting)) / (competencyWeighting + topicsWeighting)
        }
    }

    useEffect(() => {
        if (courseContentWeighting && !contentLoading && !competenciesLoading) {
            let total = calcCourseTotal( contentTotal, competencyTotal, courseContentWeighting)
            setCourseTotalLabel(total ? total.toFixed(0) + ' %' : 'N/A')
        }
    },[courseContentWeighting, contentTotal, competencyTotal, contentLoading, competenciesLoading])

    return (
        <Paper style={{padding: theme.spacing(2), margin: theme.spacing(1,0,1,0)}} elevation={10}>
            <Grid container >
                <Grid item xs={12} xl={12}>
                    <Typography variant={'h4'}>Overall Score: *</Typography>
                    { (competenciesLoading || contentLoading )
                        ? <CircularProgress color={'primary'} />
                        : <Typography color='secondary' variant={'h1'}>{courseTotalLabel}</Typography>}
                    <Typography color='secondary' variant={"subtitle1"}>* Based on tasks assessed so far</Typography>
                </Grid>
                <Grid item xs={6} xl={6}>
                    <Typography variant={'h5'}>Competency Total:</Typography>
                    { competenciesLoading
                        ? <CircularProgress color={'primary'} />
                        : <Typography color='textPrimary' variant={'h3'}>{competencyLabel}</Typography>
                    }
                </Grid>
                <Grid item xs={6} xl={6} >
                    <Typography variant={'h5'}>Content Total:</Typography>
                    { contentLoading
                        ? <CircularProgress color={'primary'} />
                        : <Typography color={'textPrimary'} variant={'h3'}>{contentLabel}</Typography>
                    }
                </Grid>

            </Grid>
        </Paper>
    )
}