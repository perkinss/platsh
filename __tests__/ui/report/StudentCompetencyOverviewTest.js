import React, { useState } from 'react'
import { createRender, createShallow } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import { ConfigurationContext } from "../../../app/javascript/context/ConfigurationContext";
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'
import TeachersStudentReport from "../../../app/javascript/components/report/TeachersStudentReport";
import StudentGradeOverview from "../../../app/javascript/components/report/StudentGradeOverview";
import StudentCompetencyOverview from "../../../app/javascript/components/report/StudentCompetencyOverview";

Enzyme.configure({ adapter: new Adapter() });

describe('<StudentCompetencyOverview />', () => {
    let render;
    let classes = {
        content: {}
    }
    let state = {
    }
    let student = {
        id: 5, name: "Ed Shearan", unique_id: "edshe@ran.com", created_at: "2019-07-07T05:07:00.505Z", updated_at: "2019-07-07T05:07:00.505Z"
    }
    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <StudentCompetencyOverview classes={classes} student={student} />
                </ConfigurationContext.Provider>
            </ThemeProvider>
                );
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should have the student details on first render', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <StudentCompetencyOverview classes={classes} student={student} />
                </ConfigurationContext.Provider>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toContain("Competency Overview")
    });

})