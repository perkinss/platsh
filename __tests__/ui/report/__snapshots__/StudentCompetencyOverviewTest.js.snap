// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudentCompetencyOverview /> should match the snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "style": "padding-top:40px",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiTypography-root MuiTypography-h5",
        },
        "children": Array [
          Node {
            "data": "Competency Overview",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h5",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
